### PR TITLE
Laboratory work #4, Pershina Olga - 21FPL1

### PR DESCRIPTION
**Pull request by OlyaP12**

### CODE REVIEW

**Comment by artyomtugaryov** [comment]
nothing to review(

File: lab_3_keywords_textrank/main.py [hunk 1]
```diff
@@ -41,6 +41,7 @@ class TextPreprocessor:
     preprocess_text(text: str) -> tuple[str, ...]:
         Produces filtered clean lowercase tokens from raw text
     """
+
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:44]: revert

File: lab_3_keywords_textrank/start.py [hunk 2]
```diff
@@ -19,7 +19,6 @@
 
 
 if __name__ == "__main__":
-
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/start.py:22]: revert

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
What about 6?8?10?

File: lab_3_keywords_textrank/start.py (hunk 2 above)
Comment (OlyaP12) [reply lab_3_keywords_textrank/start.py]: done

File: lab_3_keywords_textrank/main.py (hunk 1 above)
Comment (OlyaP12) [reply lab_3_keywords_textrank/main.py]: done

File: lab_4_summarization_textrank/main.py [hunk 3]
```diff
@@ -2,77 +2,120 @@
...
+    check_type(var, object_type)
+    for ele in var:
+        check_type(ele, element_type)
+
+
 class Sentence:
     """
     An abstraction over the real-world sentences
     """
+    _preprocessed: tuple[str, ...]
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:50]: why?

File: lab_4_summarization_textrank/main.py [hunk 4]
```diff
@@ -120,23 +178,34 @@ def __init__(self) -> None:
...
         :param tokens: a sequence of string tokens
         :return:
         """
-        pass
+        if not isinstance(tokens, tuple):
+            raise ValueError
+        if not all(isinstance(token, str) for token in tokens):
+            raise ValueError
+        new_tokens = (token for token in tokens if token not in self._word2id)
+        for idx, token in enumerate(new_tokens, 1000 + len(self._word2id)):
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:195]: last_id

File: lab_4_summarization_textrank/main.py [hunk 5]
```diff
@@ -120,23 +178,34 @@ def __init__(self) -> None:
...
+        self.last_id = 1000
 
     def _learn_indices(self, tokens: tuple[str, ...]) -> None:
         """
         Fills attributes mapping words and integer equivalents to each other
         :param tokens: a sequence of string tokens
         :return:
         """
-        pass
+        if not isinstance(tokens, tuple):
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:190]: use check_object_and_type

File: lab_4_summarization_textrank/main.py [hunk 6]
```diff
@@ -146,7 +215,11 @@ def calculate_similarity(sequence: Union[list, tuple], other_sequence: Union[lis
     :param other_sequence: a sequence of items
     :return: similarity score
     """
-    pass
+    if not isinstance(sequence, (list, tuple)) or not isinstance(other_sequence, (list, tuple)):
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:218]: .

**Review by demid5111 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**